db/models.py

```diff
@@ -45,3 +55,79 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class User(AbstractUser):
+    pass
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        to=settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE
+    )
+
+    def __str__(self) -> str:
+        return str(self.created_at)
+
+    class Meta:
+        ordering = ["-created_at"]
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(
+        to=MovieSession,
+        on_delete=models.CASCADE
+    )
+    order = models.ForeignKey(to=Order, on_delete=models.CASCADE)
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    def __str__(self) -> str:
+        return (
+            f"{self.movie_session.movie} "
+            f"{self.movie_session.show_time} "
+            f"(row: {self.row}, seat: {self.seat})"
+        )
+
+    def clean(self) -> None:
+        hall_rows = self.movie_session.cinema_hall.rows
+        hall_row_seats = self.movie_session.cinema_hall.seats_in_row
+
+        if not (1 <= self.row <= hall_rows):
+            raise ValidationError(
+                {"row": f"row number must be in available range: "
+                        f"(1, rows): (1, {hall_rows})"}
+
+            )
+
+        if not (1 <= self.seat <= hall_row_seats):
+            raise ValidationError(
+                {"seat": f"seat number must be in available range: "
+                         f"(1, seats_in_row): (1, {hall_row_seats})"}
+            )
+
+    def save(
+            self,
+            force_insert: bool = False,
+            force_update: bool = False,
+            using: str = None,
+            update_fields: str = None
+    ) -> None:
+        self.full_clean()
+
+        return super(Ticket, self).save(
+            force_insert,
+            force_update,
+            using,
+            update_fields
+        )
+
+    class Meta:
```

Comment:
I don't now, but by my opinion it will be better for readability if you put Meta right after attributes

Reply:
Yes, I agree. Somehow thought that typical order is 'attrs -> methods -> meta'.
Just checked. As far as seen, the best practice is considered 'attrs -> meta -> methods'